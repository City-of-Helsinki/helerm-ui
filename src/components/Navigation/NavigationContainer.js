import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import { fetchNavigation, setNavigationVisibility } from './reducer';

import { itemById } from '../../utils/helpers';

import Navigation from './Navigation';

const mapDispatchToProps = (dispatch) => {
  return {
    fetchNavigation: bindActionCreators(fetchNavigation, dispatch),
    push: (path) => dispatch(push(path)),
    setNavigationVisibility: bindActionCreators(setNavigationVisibility, dispatch)
  };
};

const mapStateToProps = (state, ownProps) => {
  const { items } = state.navigation;
  const { selectedTOS, classification } = state;
  const tos = selectedTOS.classification ? itemById(items, selectedTOS.classification) : classification.id ? itemById(items, classification.id) : null;
  const tosPath = ownProps.tosPath ? ownProps.tosPath : tos ? tos.path : [];

  return {
    tosPath,
    is_open: state.navigation.is_open,
    isFetching: state.navigation.isFetching,
    items,
    selectedTOS
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Navigation);
