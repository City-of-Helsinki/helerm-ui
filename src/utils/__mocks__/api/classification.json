[
  {
    "id": "test-classification-health-guidance-001",
    "created_at": "2017-06-07T05:48:15.195700Z",
    "modified_at": "2017-12-27T10:39:28.775708Z",
    "modified_by": "Test User",
    "version": 2,
    "state": "approved",
    "valid_from": null,
    "valid_to": null,
    "code": "00 00 00 00",
    "title": "Terveysneuvonta",
    "parent": null,
    "description": "",
    "description_internal": "",
    "related_classification": "",
    "additional_information": "",
    "function_allowed": true,
    "version_history": [
      {
        "state": "draft",
        "version": 1,
        "modified_at": "2017-12-26T10:39:28.775708Z",
        "valid_from": null,
        "valid_to": null,
        "modified_by": "Test User"
      },
      {
        "version": 2,
        "state": "approved",
        "modified_at": "2017-12-27T10:39:28.775708Z",
        "modified_by": "Test User",
        "valid_from": null,
        "valid_to": null
      }
    ],
    "function": "test-function-election-001",
    "function_state": "approved",
    "function_attributes": {
      "PersonalData": "Sisältää henkilötietoja",
      "PublicityClass": "Julkinen",
      "RetentionPeriod": "5",
      "RetentionReason": "Test retention reason for consistent testing",
      "InformationSystem": "Test Information System",
      "SocialSecurityNumber": "Ei sisällä henkilötunnusta"
    },
    "function_version": 14,
    "function_valid_from": "2017-11-13",
    "function_valid_to": "2018-11-27"
  },
  {
    "id": "test-classification-education-001",
    "created_at": "2017-07-07T05:48:15.195700Z",
    "modified_at": "2017-12-28T10:39:28.775708Z",
    "modified_by": "Test User",
    "version": 1,
    "state": "approved",
    "valid_from": null,
    "valid_to": null,
    "code": "07 01 00",
    "title": "Koulutuspalvelut",
    "parent": null,
    "description": "",
    "description_internal": "",
    "related_classification": "",
    "additional_information": "",
    "function_allowed": true,
    "version_history": [
      {
        "state": "draft",
        "version": 1,
        "modified_at": "2017-12-26T10:39:28.775708Z",
        "valid_from": null,
        "valid_to": null,
        "modified_by": "Test User"
      }
    ],
    "function": "07 01 00",
    "function_state": "approved",
    "function_attributes": {
      "PersonalData": "Sisältää henkilötietoja",
      "PublicityClass": "Julkinen",
      "RetentionPeriod": "10",
      "RetentionReason": "Test education retention reason",
      "InformationSystem": "Test Education System",
      "SocialSecurityNumber": "Ei sisällä henkilötunnusta"
    },
    "function_version": 8,
    "function_valid_from": "2017-12-01",
    "function_valid_to": "2019-01-15"
  },
  {
    "id": "test-classification-education-child-001",
    "created_at": "2017-08-07T05:48:15.195700Z",
    "modified_at": "2017-12-29T10:39:28.775708Z",
    "modified_by": "Test User",
    "version": 1,
    "state": "approved",
    "valid_from": null,
    "valid_to": null,
    "code": "07 01 01",
    "title": "Koulutuksen järjestäminen",
    "parent": "test-classification-education-001",
    "description": "",
    "description_internal": "",
    "related_classification": "",
    "additional_information": "",
    "function_allowed": true,
    "version_history": [
      {
        "state": "draft",
        "version": 1,
        "modified_at": "2017-12-29T10:39:28.775708Z",
        "valid_from": null,
        "valid_to": null,
        "modified_by": "Test User"
      }
    ],
    "function": "07 01 01",
    "function_state": "approved",
    "function_attributes": {
      "PersonalData": "Sisältää erityisiä henkilötietoja",
      "PublicityClass": "Osittain salassa pidettävä",
      "RetentionPeriod": "5",
      "RetentionReason": "Test education organization reason",
      "InformationSystem": "Test Education Organization System",
      "SocialSecurityNumber": "Sisältää henkilötunnuksen"
    },
    "function_version": 3,
    "function_valid_from": "2018-01-01",
    "function_valid_to": "2019-01-31"
  },
  {
    "id": "test-classification-draft-001",
    "created_at": "2023-01-01T00:00:00.000000Z",
    "modified_at": "2023-01-01T10:30:00.000000Z",
    "modified_by": "Draft User",
    "version": 1,
    "state": "draft",
    "valid_from": null,
    "valid_to": null,
    "code": "08 01 00 00",
    "title": "Luonnos tehtäväluokka",
    "parent": null,
    "description": "Test draft classification",
    "description_internal": "Internal draft description",
    "related_classification": "",
    "additional_information": "",
    "function_allowed": true,
    "version_history": [
      {
        "state": "draft",
        "version": 1,
        "modified_at": "2023-01-01T10:30:00.000000Z",
        "valid_from": null,
        "valid_to": null,
        "modified_by": "Draft User"
      }
    ],
    "function": "test-function-draft-001",
    "function_state": "draft",
    "function_attributes": {
      "PersonalData": "Ei sisällä henkilötietoja",
      "PublicityClass": "Julkinen",
      "RetentionPeriod": "3",
      "RetentionReason": "Test draft retention reason",
      "InformationSystem": "Test Draft System",
      "SocialSecurityNumber": "Ei sisällä henkilötunnusta"
    },
    "function_version": 1,
    "function_valid_from": "2023-01-01",
    "function_valid_to": null
  },
  {
    "id": "test-classification-sent-for-review-001",
    "created_at": "2023-01-02T00:00:00.000000Z",
    "modified_at": "2023-01-02T14:15:00.000000Z",
    "modified_by": "Review User",
    "version": 1,
    "state": "sent_for_review",
    "valid_from": null,
    "valid_to": null,
    "code": "09 01 00 00",
    "title": "Tarkastuksessa oleva tehtäväluokka",
    "parent": null,
    "description": "Test classification sent for review",
    "description_internal": "Internal review description",
    "related_classification": "",
    "additional_information": "",
    "function_allowed": true,
    "version_history": [
      {
        "state": "draft",
        "version": 1,
        "modified_at": "2023-01-02T10:00:00.000000Z",
        "valid_from": null,
        "valid_to": null,
        "modified_by": "Review User"
      }
    ],
    "function": "test-function-review-001",
    "function_state": "sent_for_review",
    "function_attributes": {
      "PersonalData": "Sisältää henkilötietoja",
      "PublicityClass": "Osittain salassa pidettävä",
      "RetentionPeriod": "7",
      "RetentionReason": "Test review retention reason",
      "InformationSystem": "Test Review System",
      "SocialSecurityNumber": "Ei sisällä henkilötunnusta"
    },
    "function_version": 1,
    "function_valid_from": "2023-01-02",
    "function_valid_to": null
  },
  {
    "id": "test-classification-waiting-approval-001",
    "created_at": "2023-01-03T00:00:00.000000Z",
    "modified_at": "2023-01-03T16:45:00.000000Z",
    "modified_by": "Approval User",
    "version": 1,
    "state": "waiting_for_approval",
    "valid_from": null,
    "valid_to": null,
    "code": "10 01 00 00",
    "title": "Hyväksyntää odottava tehtäväluokka",
    "parent": null,
    "description": "Test classification waiting for approval",
    "description_internal": "Internal approval description",
    "related_classification": "",
    "additional_information": "",
    "function_allowed": true,
    "version_history": [
      {
        "state": "draft",
        "version": 1,
        "modified_at": "2023-01-03T10:00:00.000000Z",
        "valid_from": null,
        "valid_to": null,
        "modified_by": "Approval User"
      }
    ],
    "function": "test-function-waiting-001",
    "function_state": "waiting_for_approval",
    "function_attributes": {
      "PersonalData": "Sisältää henkilötietoja",
      "PublicityClass": "Julkinen",
      "RetentionPeriod": "10",
      "RetentionReason": "Test waiting approval retention reason",
      "InformationSystem": "Test Approval System",
      "SocialSecurityNumber": "Sisältää henkilötunnuksen"
    },
    "function_version": 1,
    "function_valid_from": "2023-01-03",
    "function_valid_to": null
  }
]
