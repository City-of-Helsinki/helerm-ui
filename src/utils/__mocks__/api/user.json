{
  "uuid": "test-user-001",
  "username": "test.user",
  "first_name": "Test",
  "last_name": "User",
  "permissions": [
    "can_edit",
    "can_review",
    "can_approve",
    "can_view_modified_by",
    "add_bulkupdate",
    "approve_bulkupdate",
    "change_bulkupdate",
    "delete_bulkupdate",
    "view_bulkupdate"
  ]
}
