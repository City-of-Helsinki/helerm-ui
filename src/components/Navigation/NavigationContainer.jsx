/* eslint-disable camelcase */
/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from '@reduxjs/toolkit';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';
import { isEmpty } from 'lodash';

import { fetchNavigation, setNavigationVisibility } from './reducer';
import { itemById } from '../../utils/helpers';
import Navigation from './Navigation';
import withRouter from '../hoc/withRouter';

const NavigationContainer = ({
  attributeTypes,
  fetchNavigationFn,
  isFetching,
  isUser,
  is_open,
  items,
  itemsTimestamp,
  onLeafMouseClick,
  pushFn,
  setNavigationVisibilityFn,
  tosPath,
  location,
}) => (
  <Navigation
    attributeTypes={attributeTypes}
    fetchNavigation={fetchNavigationFn}
    isFetching={isFetching}
    isUser={isUser}
    is_open={is_open}
    items={items}
    itemsTimestamp={itemsTimestamp}
    onLeafMouseClick={onLeafMouseClick}
    push={pushFn}
    setNavigationVisibility={setNavigationVisibilityFn}
    tosPath={tosPath}
    location={location}
  />
);

const mapDispatchToProps = (dispatch) => ({
  fetchNavigationFn: bindActionCreators(fetchNavigation, dispatch),
  pushFn: (path) => dispatch(push(path)),
  setNavigationVisibilityFn: bindActionCreators(setNavigationVisibility, dispatch),
});

const getTOS = (selectedTOS, items, classification) => {
  if (selectedTOS.classification) {
    return itemById(items, selectedTOS.classification.id);
  }

  if (classification.id) {
    return itemById(items, classification.id);
  }

  return null;
};

const getTOSPath = (ownProps, tos) => {
  if (ownProps.tosPath) {
    return ownProps.tosPath;
  }

  if (tos) {
    return tos.path;
  }

  return [];
};

const mapStateToProps = (state, ownProps) => {
  const { items, timestamp } = state.navigation;
  const { attributeTypes } = state.ui;
  const { selectedTOS, classification } = state;

  const tos = getTOS(selectedTOS, items, classification);
  const tosPath = getTOSPath(ownProps, tos);

  return {
    attributeTypes,
    tosPath,
    is_open: state.navigation.is_open,
    isFetching: state.navigation.isFetching,
    isUser: !isEmpty(state.user.data),
    items, // TODO: Unhack this when Navigation doesn't mutate state
    itemsTimestamp: timestamp,
    selectedTOS,
  };
};

NavigationContainer.propTypes = {
  attributeTypes: PropTypes.object,
  fetchNavigationFn: PropTypes.func.isRequired,
  isFetching: PropTypes.bool,
  isUser: PropTypes.bool,
  is_open: PropTypes.bool.isRequired,
  // One does not simply mutate props unless one is Navigation and the prop is `items`.
  // Sorry, didn't find out where the devil is doing the mutations :'(
  items: PropTypes.array.isRequired,
  itemsTimestamp: PropTypes.string,
  onLeafMouseClick: PropTypes.func,
  pushFn: PropTypes.func.isRequired,
  setNavigationVisibilityFn: PropTypes.func.isRequired,
  tosPath: PropTypes.array.isRequired,
  location: PropTypes.object.isRequired,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(NavigationContainer));
