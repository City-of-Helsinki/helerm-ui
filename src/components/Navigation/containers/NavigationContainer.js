import { connect } from 'react-redux';

import {
  requestNavigation,
  receiveNavigation,
  setNavigationVisibility,
  fetchNavigation
} from '../modules/navigation';

import Navigation from '../components/Navigation';

const mapDispatchToProps = {
  requestNavigation,
  receiveNavigation,
  setNavigationVisibility,
  fetchNavigation
};

const mapStateToProps = (state) => {
  return {
    isFetching: state.navigation.isFetching,
    items: state.navigation.items,
    is_open: state.navigation.is_open
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Navigation);
