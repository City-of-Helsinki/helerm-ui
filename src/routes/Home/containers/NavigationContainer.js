import { connect } from 'react-redux';
import { getNavigationMenuItems } from '../modules/navigation';
import Navigation from '../components/Navigation';

const mapDispatchToProps = {
  getNavigationMenuItems
};

const mapStateToProps = (state) => {
  return {
    navigationMenuItems : state.navigation.navigationMenuItems
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(Navigation);
