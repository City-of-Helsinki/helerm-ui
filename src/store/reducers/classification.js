import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';

import api from '../../utils/api';
import { fetchNavigationThunk } from './navigation';

export const initialState = {
  id: null,
  function: null,
  code: null,
  parent: null,
  title: null,
  description: null,
  description_internal: null,
  related_classification: null,
  additional_information: null,
  state: null,
  created_at: null,
  modified_at: null,
  isFetching: false,

  version: null,
  version_history: [],
  function_allowed: false,
  function_version: null,
  error: null,
};

export const fetchClassificationThunk = createAsyncThunk(
  'classification/fetchClassification',
  async ({ id, params = {} }, { rejectWithValue }) => {
    try {
      const response = await api.get(`classification/${id}`, params);

      if (!response.ok) {
        throw new URIError(response.statusText);
      }

      const data = await response.json();

      return { ...initialState, ...data };
    } catch (error) {
      return rejectWithValue(error instanceof Error ? error.message : 'Failed to fetch classification');
    }
  },
);

export const createTosThunk = createAsyncThunk(
  'classification/createTos',
  async (_, { dispatch, getState, rejectWithValue }) => {
    try {
      const classification = { ...getState().classification };
      const newTos = {
        classification: { id: classification.id, version: classification.version },
      };

      const response = await api.post('function', newTos);

      if (!response.ok) {
        throw Error(response.statusText);
      }

      const json = await response.json();

      const { includeRelated } = getState().navigation;

      dispatch(fetchNavigationThunk({ includeRelated }));

      return json;
    } catch (error) {
      return rejectWithValue(error instanceof Error ? error.message : 'Failed to create TOS');
    }
  },
);

const classificationSlice = createSlice({
  name: 'classification',
  initialState,
  reducers: {
    clearClassification: () => initialState,
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchClassificationThunk.pending, (state) => {
        state.isFetching = true;
        state.error = null;
      })
      .addCase(fetchClassificationThunk.fulfilled, (state, action) => {
        return { ...state, ...action.payload, isFetching: false, error: null };
      })
      .addCase(fetchClassificationThunk.rejected, (state, action) => {
        state.isFetching = false;
        state.error = action.payload || 'Failed to fetch classification';
      })

      .addCase(createTosThunk.pending, (state) => {
        state.isFetching = true;
        state.error = null;
      })
      .addCase(createTosThunk.fulfilled, (state, action) => {
        state.isFetching = false;
        state.function = action.payload.id;
        state.function_version = action.payload.version;
        state.error = null;
      })
      .addCase(createTosThunk.rejected, (state, action) => {
        state.isFetching = false;
        state.error = action.payload || 'Failed to create TOS';
      });
  },
  selectors: {
    classificationSelector: (state) => state,
    idSelector: (state) => state.id,
    functionSelector: (state) => state.function,
    codeSelector: (state) => state.code,
    titleSelector: (state) => state.title,
    descriptionSelector: (state) => state.description,
    stateSelector: (state) => state.state,
    versionSelector: (state) => state.version,
    versionHistorySelector: (state) => state.version_history,
    descriptionInternalSelector: (state) => state.description_internal,
    relatedClassificationSelector: (state) => state.related_classification,
    additionalInformationSelector: (state) => state.additional_information,
    parentSelector: (state) => state.parent,
    isFetchingSelector: (state) => state.isFetching,
    errorSelector: (state) => state.error,
    hasFunctionSelector: (state) => !!state.function,
    canCreateTosSelector: (state) => !state.function && state.function_allowed,
    createdAtSelector: (state) => state.created_at,
    modifiedAtSelector: (state) => state.modified_at,
    functionVersionSelector: (state) => state.function_version,
  },
});

export const { clearClassification } = classificationSlice.actions;

export const {
  classificationSelector,
  idSelector,
  functionSelector,
  codeSelector,
  titleSelector,
  descriptionSelector,
  stateSelector,
  versionSelector,
  versionHistorySelector,
  descriptionInternalSelector,
  relatedClassificationSelector,
  additionalInformationSelector,
  parentSelector,
  isFetchingSelector,
  errorSelector,
  hasFunctionSelector,
  canCreateTosSelector,
  createdAtSelector,
  modifiedAtSelector,
  functionVersionSelector,
} = classificationSlice.selectors;

export default classificationSlice.reducer;
