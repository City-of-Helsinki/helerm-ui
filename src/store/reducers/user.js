import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import { get } from 'lodash';

import api, { Unauthorized } from '../../utils/api';
import { USER_LOGIN_STATUS } from '../../constants';

export const initialState = {
  data: null,
  isFetching: false,
  status: USER_LOGIN_STATUS.NONE,
  error: null,
};

export const retrieveUserFromSession = createAsyncThunk(
  'user/retrieveUserFromSession',
  async ({ id, token }, { dispatch, rejectWithValue }) => {
    try {
      if (!id || !token) {
        dispatch(loginSlice.actions.setLoginStatus(USER_LOGIN_STATUS.NONE));

        return rejectWithValue('No ID or token provided');
      }

      dispatch(loginSlice.actions.setLoginStatus(USER_LOGIN_STATUS.INITIALIZING));

      const url = `user/${id}`;
      const response = await api.get(url, {}, {}, token);

      if (response.status === 401) {
        throw new Unauthorized(url);
      }

      const json = await response.json();
      const permissions = get(json, 'permissions', []);

      const userWithPermissions = {
        id,
        firstName: json.first_name,
        lastName: json.last_name,
        permissions,
      };

      dispatch(loginSlice.actions.setLoginStatus(USER_LOGIN_STATUS.AUTHORIZED));

      return userWithPermissions;
    } catch (error) {
      dispatch(loginSlice.actions.setLoginStatus(USER_LOGIN_STATUS.UNAUTHORIZED));

      return rejectWithValue(error instanceof Error ? error.message : 'Failed to retrieve user data');
    }
  },
);

export const logoutUserThunk = createAsyncThunk('user/logout', async (_, { dispatch }) => {
  dispatch(loginSlice.actions.setLoginStatus(USER_LOGIN_STATUS.NONE));

  return null;
});

export const initializeLoginCallbackThunk = createAsyncThunk(
  'user/initializeLoginCallback',
  async (_, { dispatch }) => {
    dispatch(loginSlice.actions.setLoginStatus(USER_LOGIN_STATUS.INITIALIZING));

    return null;
  },
);

export const handleLoginCallbackErrorThunk = createAsyncThunk(
  'user/handleLoginCallbackError',
  async (error, { dispatch, rejectWithValue }) => {
    dispatch(loginSlice.actions.setLoginStatus(USER_LOGIN_STATUS.UNAUTHORIZED));

    return rejectWithValue(error instanceof Error ? error.message : String(error));
  },
);

const loginSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    clearUserData: () => initialState,
    setLoginStatus: (state, action) => {
      state.status = action.payload;
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(retrieveUserFromSession.pending, (state) => {
        state.isFetching = true;
        state.error = null;
      })
      .addCase(retrieveUserFromSession.fulfilled, (state, action) => {
        state.data = action.payload;
        state.isFetching = false;
        state.error = null;
      })
      .addCase(retrieveUserFromSession.rejected, (state, action) => {
        state.data = {};
        state.isFetching = false;
        state.error = action.payload || 'Failed to retrieve user data';
      })

      .addCase(logoutUserThunk.fulfilled, () => {
        return initialState;
      })

      .addCase(initializeLoginCallbackThunk.fulfilled, (state) => {
        state.status = USER_LOGIN_STATUS.INITIALIZING;
      })

      .addCase(handleLoginCallbackErrorThunk.fulfilled, (state) => {
        state.status = USER_LOGIN_STATUS.UNAUTHORIZED;
        state.data = null;
      });
  },
  selectors: {
    userDataSelector: (state) => state.data,
    userPermissionsSelector: (state) => state.data?.permissions || [],
    loginStatusSelector: (state) => state.status,
    isFetchingSelector: (state) => state.isFetching,
    errorSelector: (state) => state.error,
    isAuthorizedSelector: (state) => state.status === USER_LOGIN_STATUS.AUTHORIZED,
    isAuthenticatedSelector: (state) =>
      state.status === USER_LOGIN_STATUS.AUTHORIZED || state.status === USER_LOGIN_STATUS.INITIALIZING,
    hasPermissionSelector: (state, permission) => (state.data?.permissions || []).includes(permission),
  },
});

export const { clearUserData, setLoginStatus: loginStatus } = loginSlice.actions;

export const {
  userDataSelector,
  userPermissionsSelector,
  loginStatusSelector,
  isFetchingSelector,
  errorSelector,
  isAuthorizedSelector,
  isAuthenticatedSelector,
  hasPermissionSelector,
} = loginSlice.selectors;

export default loginSlice.reducer;
