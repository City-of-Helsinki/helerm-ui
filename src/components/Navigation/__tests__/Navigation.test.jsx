import React from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';

import renderWithProviders from '../../../utils/renderWithProviders';
import NavigationContainer from '../NavigationContainer';
import attributeRules from '../../../utils/mocks/attributeRules.json';
import validTOS from '../../../utils/mocks/validTOS.json';
import classification from '../../../utils/mocks/classification.json';
import user from '../../../utils/mocks/user.json';

const renderComponent = (history) =>
  renderWithProviders(
    <Router history={history}>
      <NavigationContainer />
    </Router>,
    {
      history,
      preloadedState: {
        selectedTOS: validTOS,
        navigation: {
          timestamp: '123456789',
          items: [validTOS],
          classification,
          is_open: false,
          isFetching: false,
        },
        ui: { attributeTypes: attributeRules },
        user: {
          data: user,
        },
      },
    },
  );

describe('<Navigation />', () => {
  it('should render correctly', () => {
    const history = createBrowserHistory();

    const { container } = renderComponent(history);

    expect(container).toMatchSnapshot();
  });
});
