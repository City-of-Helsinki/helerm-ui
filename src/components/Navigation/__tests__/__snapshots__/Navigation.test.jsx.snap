// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render correctly 1`] = `
<div>
  <div
    class="helerm-navigation"
  >
    <div
      class="navigation-menu"
      id="navigation-menu"
    >
      <div
        style="min-height: 0px;"
      >
        <div
          class=" "
          style="transform: translateZ(0);"
        >
          <div
            class="navigation-header clearfix"
          >
            <button
              class="pull-left nav-button"
              type="button"
            >
              <span
                aria-hidden="true"
                class="fa-solid fa-list"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
